Update to not release test bundles that are in multiple projects

jhanders34 opened the pull request:
- When 2 or more test projects end up having test bundles with the same name, there can be errors during the build when both projects try to release the same bundle name and it already was released.  There is no need to actually release test bundles to the bnd repository.  This PR updates to not release the bundles whose names are shared across multiple test projects. 
- In the future it would be good to update all test bundle bnd files to not release, but this is a stop gap in order to prevent people from getting build errors when doing gradle with parallel enabled.
- Also updated bnd.bnd files to remove SymbolicName settings when the projects has sub bundles and set the symbolic name in the sub bundles.

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

jhanders34 committed:
Update to not release test bundles that are in multiple projects

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=96971692-753c-4d0c-8837-0c2ee51c60ca) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

fritze2 reviewed (dismissed):
~Why was Bundle-Name, Bundle-SymbolicName, and Bundle-Description removed in some cases?~ I see now, when it has `-sub` bundles.

How do you know whether a bundle is a test bundle to know that `releaserepo:` needed to be added to disable it from being released? It is a simple change, so assuming it passes the build it looks good. I just wonder if there are test bundles missed and how you know you got them all.

jhanders34 commented:
> How do you know whether a bundle is a test bundle to know that `releaserepo:` needed to be added to disable it from being released? It is a simple change, so assuming it passes the build it looks good. I just wonder if there are test bundles missed and how you know you got them all.

I looked at all symbolic names and if it was duplicated I changed not to release the bundle.  They are test bundles if in test projects.  The non test ones I just removed the bnd.bnd one if it was already on original.bnd.

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=81fa66da-3558-43c3-8b8a-1dbefff3c055) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._
